File: src/main/csharp/Core/Transformer.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace App.Core
{
    public static class Transformer
    {
        public static int StringToInt(string input) => int.Parse(input);
        
        public static long StringToLong(string input) => long.Parse(input);
        
        public static double StringToDouble(string input) => double.Parse(input);
        
        public static bool StringToBool(string input) => bool.Parse(input);
        
        public static T GetFirst<T>(List<T> list) where T : class
        {
            return list.Count == 0 ? null : list[0];
        }
        
        public static T GetLast<T>(List<T> list) where T : class
        {
            return list.Count == 0 ? null : list[list.Count - 1];
        }
        
        public static List<long> ExtractIds<T>(List<T> items) where T : IHasId
        {
            return items.Select(x => x.Id).ToList();
        }
        
        public static List<int> ExtractStatuses<T>(List<T> items) where T : IHasStatus
        {
            return items.Select(x => x.Status).ToList();
        }
    }
    
    public interface IHasId
    {
        long Id { get; }
    }
    
    public interface IHasStatus
    {
        int Status { get; }
    }
}

File: src/main/java/com/app/core/Transformer.java
package com.app.core;

import java.util.List;
import java.util.stream.Collectors;

public class Transformer {
    
    public static int stringToInt(String input) {
        return Integer.parseInt(input);
    }
    
    public static long stringToLong(String input) {
        return Long.parseLong(input);
    }
    
    public static double stringToDouble(String input) {
        return Double.parseDouble(input);
    }
    
    public static boolean stringToBoolean(String input) {
        return Boolean.parseBoolean(input);
    }
    
    public static <T> T getFirst(List<T> list) {
        return list.isEmpty() ? null : list.get(0);
    }
    
    public static <T> T getLast(List<T> list) {
        return list.isEmpty() ? null : list.get(list.size() - 1);
    }
    
    public static List<Long> extractIds(List<? extends HasId> items) {
        return items.stream().map(HasId::getId).collect(Collectors.toList());
    }
    
    public static List<Integer> extractStatuses(List<? extends HasStatus> items) {
        return items.stream().map(HasStatus::getStatus).collect(Collectors.toList());
    }
    
    public interface HasId {
        long getId();
    }
    
    public interface HasStatus {
        int getStatus();
    }
}

